src/lein_monolith/task/fingerprint.clj

```diff
@@ -188,9 +218,19 @@
   "Create a stateful context to use for fingerprinting operations."
   [monolith subprojects]
   (let [dep-map (dep/dependency-map subprojects)
-        initial (read-fingerprints-file monolith)
-        cache (atom {})]
-    {:monolith monolith
+        root (:root monolith)
+        initial (read-fingerprints-file root)
+        cache (atom {})
+        subprojects (into {}
+                          (map
+                            (fn inherit-profiles
+                              [[k subproject]]
+                              [k
+                               (update subproject :profiles merge
+                                       (plugin/build-inherited-profiles
+                                         monolith subproject))]))
```

Comment:
Could be clearer to pull out this function:
```clojure
inherit-profiles (fn inherit-profiles
                   [subproject]
                   (update subproject :profiles merge
                           (plugin/build-inherited-profiles
                             monolith subproject)))
subprojects (into {} (map (juxt key inherit-profiles)) subprojects)
```

Reply:
I think the latter expression would have to be
```clojure
subprojects (into {} (map (juxt key (comp inherit-profiles val))) subprojects)
```
which feels like less of a clarity win.